src/main/java/seedu/duke/models/schema/Schedule.java

```diff
@@ -286,6 +287,109 @@ public void deleteModule(String module) throws FailPrereqException, MissingModul
         //
     }
 
+    public void shiftModule(String module, int targetSem) throws IllegalArgumentException,
+            FailPrereqException, MandatoryPrereqException, MissingModuleException, InvalidObjectException {
```

Comment:
I would recommend refactoring this function as it is 80-100 loc currently. Here's the code quality advice for these situations! 
<img width="703" alt="image" src="https://github.com/AY2324S1-CS2113-T17-4/tp/assets/110933149/642829c5-bfff-41f9-b9e3-8b9f6fa1da21">